Assets/MixedRealityToolkit.Providers/XRSDK/Editor/ConfigurationChecker/XRSDKConfigurationChecker.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK
+{
+    /// <summary>
+    /// Class to perform checks for configuration checks for the UnityAR provider.
+    /// </summary>
+    [InitializeOnLoad]
+    static class XRSDKConfigurationChecker
+    {
+        private const string AsmDefFileName = "Microsoft.MixedReality.Toolkit.Providers.XRSDK.asmdef";
+        private const string XRManagementReference = "Unity.XR.Management";
+        private const string ARSubsystemsReference = "Unity.XR.ARSubsystems";
+
+#if UNITY_2019_3_OR_NEWER
+        private static readonly VersionDefine XRManagementDefine = new VersionDefine("com.unity.xr.management", "", "XR_MANAGEMENT_ENABLED");
+        private static readonly VersionDefine ARSubsystemsDefine = new VersionDefine("com.unity.xr.arsubsystems", "", "ARSUBSYSTEMS_ENABLED");
+#endif // UNITY_2019_3_OR_NEWER
+
+        static XRSDKConfigurationChecker()
+        {
+            UpdateAsmDef();
+        }
+
+        /// <summary>
+        /// Updates the assembly definition to contain the appropriate references based on the Unity version.
+        /// </summary>
+        /// <remarks>
+        /// Versions of Unity may have different factorings of components. To address this, the UpdateAsmDef
+        /// method conditionally compiles for the version currently in use.
+        /// To ensure proper compilation on each Unity version, the following steps are performed:
+        /// - Load the Microsoft.MixedReality.Toolkit.Providers.XRSDK.asmdef file
+        /// - If Unity 2018: nothing
+        /// - If Unity 2019 and newer: Unity.XR.Management, 
```

Comment:
nit: dangling sentence... ends in a ','. is something missing?

Reply:
Whoops!

Reply:
Updated